yarl/_url.py

```diff
@@ -1023,6 +1024,16 @@ def update_query(self, *args, **kwargs):
             self._val._replace(query=self._get_str_query(query) or ""), encoded=True
         )
 
+    def without_query_params(self, *keys: str) -> "URL":
+        """Remove some keys from query part and return new URL."""
+        valid_keys = set(keys) & set(self.query.keys())
+        if not valid_keys:
+            return self
+        editable_query = cast(MultiDict, self.query.copy())
```

Comment:
I have to do `self.query.copy()` because `self.query` is immutable, also `.copy()` returns a shallow copy ([ref](https://multidict.aio-libs.org/en/stable/multidict.html#multidict.MultiDictProxy.copy), it doesn't waste memory much.

The `cast(MultiDict, self.query.copy())` doesn't create new object. It is just for type checking. The returned value is exact the 2nd argument.

Reply:
This should do: https://github.com/aio-libs/yarl/pull/898/files#r1414808748